### PR TITLE
fix infinite sherpa loop

### PR DESCRIPTION
**Pull request by covarell**
This is a fix of #9765 which was reverted in #11073 because of infinite loops in sherpa sample observed in IBs.

Tested locally on problematic samples, observed sensible distributions in DQM histos.

### CODE REVIEW

**Comment by deguio** [comment]
please test

**Comment by covarell** [comment]
+1

File: Validation/EventGenerator/interface/BasicHepMCValidation.h [hunk 1]
```diff
@@ -43,77 +45,156 @@ class BasicHepMCValidation : public DQMEDAnalyzer{
...
+      }
+
+      ~ParticleMonitor(){};
+      
+      bool Fill(const HepMC::GenParticle* p, double weight){
+	if(p->pdg_id()==pdgid){
+	  if(isFirst(p)){
+	    p_init->Fill(log10(p->momentum().rho()),weight);
+	    eta_init->Fill(p->momentum().eta(),weight);
+	    std::cout << "Hi, I am here" << std::endl;
```
Comment (slava77) [inline Validation/EventGenerator/interface/BasicHepMCValidation.h]: this generates about 600 lines per event  in ttbar

**Comment by slava77** [comment]
@covarell @davidlange6 
take a note
https://cmssdt.cern.ch/SDT/jenkins-artifacts/pull-request-integration/PR-11122/8318/runTheMatrix-results/25.0_TTbar+TTbar+DIGI+RECOAlCaCalo+HARVEST+ALCATT/step3_TTbar+TTbar+DIGI+RECOAlCaCalo+HARVEST+ALCATT.log

**Comment by covarell** [comment]
Oops, of course this was only there for debugging... Will open a new PR

**Pull request merged**